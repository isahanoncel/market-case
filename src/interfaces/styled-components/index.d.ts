// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft' | 'Check';

export type Colors =
  | 'primary'
  | 'gray'
  | 'secondary'
  | 'white'
  | 'background'
  | 'transparent'
  | 'innerBorder'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
