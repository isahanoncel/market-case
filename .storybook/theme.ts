import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
