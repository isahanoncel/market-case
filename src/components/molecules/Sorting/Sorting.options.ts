const options = [
  {
    type: 'price_asc',
    text: 'Price low to high',
  },
  {
    type: 'price_desc',
    text: 'Price high to low',
  },
  {
    type: 'added_asc',
    text: 'New to old',
  },
  {
    type: 'added_desc',
    text: 'Old to new',
  },
];

export default options;
