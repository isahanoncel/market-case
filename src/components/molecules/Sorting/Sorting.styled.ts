import styled from 'styled-components';
import StyledHelper from '../../../constants/styles/Helper';

export const SortingWrapper = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  box-shadow: 0px 6px 24px rgba(93, 62, 188, 0.04);
  border-radius: 2px;
  margin-top: 12px;
  padding: 24px;
  max-height: 184px;
  overflow-y: scroll;
`;

export const SortingItem = styled.div`
  cursor: pointer;
  margin-bottom: 18px;
  ${StyledHelper.CenterContentVertical}
`;

export const SortingRadio = styled.div<{ checked: boolean }>`
  ${StyledHelper.CenterContent}
  width: 22px;
  height: 22px;
  border: 2px solid ${({ theme }) => theme.colors.innerBorder};
  border-radius: 22px;
  margin: 0 12px 0 0;
  ${(props) => props.checked && `border-color:${props.theme.colors.primary}`};
`;
