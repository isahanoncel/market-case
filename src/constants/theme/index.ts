import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
