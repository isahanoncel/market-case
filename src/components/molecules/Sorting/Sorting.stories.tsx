import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Sorting from '.';

export default {
  title: 'Molecules/Sorting',
  component: Sorting,
} as ComponentMeta<typeof Sorting>;

export const Default: ComponentStory<typeof Sorting> = (args) => (
  <Sorting {...args} />
);
