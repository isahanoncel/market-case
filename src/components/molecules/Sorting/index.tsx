import React, { FC, useCallback, useMemo, useState } from 'react';
import Flex from '../../atoms/Flex';
import Label from '../../atoms/Label';
import * as S from './Sorting.styled';
import options from './Sorting.options';

const Sorting: FC = () => {
  const [selectedSorting, setSelectedSorting] = useState<string>('');

  const handleChangeSorting = useCallback(
    (item: string) => {
      setSelectedSorting(item);
    },
    [selectedSorting],
  );

  const renderOptions = useMemo(() => {
    return options.map((item) => (
      <S.SortingItem
        key={item.type}
        onClick={() => handleChangeSorting(item.type)}
      >
        <S.SortingRadio checked={selectedSorting === item.type}>
          {item.type === selectedSorting && (
            <svg
              width="10"
              height="7"
              viewBox="0 0 10 7"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 1L3.5 6L1 3.72727"
                stroke="#1EA4CE"
                strokeWidth="1.2"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </svg>
          )}
        </S.SortingRadio>
        <Label color="black600" weight={400} size={14} lineHeight={18}>
          {item.text}
        </Label>
      </S.SortingItem>
    ));
  }, [selectedSorting]);

  return (
    <Flex flexDirection="column">
      <Label size={13} weight={600} color="gray">
        Sorting
      </Label>

      <S.SortingWrapper>{renderOptions}</S.SortingWrapper>
    </Flex>
  );
};

export default Sorting;
