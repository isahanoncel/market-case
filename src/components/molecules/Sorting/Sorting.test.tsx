import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Sorting from '.';
import theme from '../../../constants/theme';

test('Sorting snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Sorting />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
