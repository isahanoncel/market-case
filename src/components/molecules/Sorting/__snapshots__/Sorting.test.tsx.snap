// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting snapshot testing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-size: 13px;
  line-height: 14px;
  color: #697488;
  font-weight: 600;
  margin: 0;
}

.c5 {
  font-size: 14px;
  line-height: 18px;
  color: #525252;
  font-weight: 400;
  margin: 0;
}

.c2 {
  background-color: #fff;
  box-shadow: 0px 6px 24px rgba(93,62,188,0.04);
  border-radius: 2px;
  margin-top: 12px;
  padding: 24px;
  max-height: 184px;
  overflow-y: scroll;
}

.c3 {
  cursor: pointer;
  margin-bottom: 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 22px;
  height: 22px;
  border: 2px solid #DFDEE2;
  border-radius: 22px;
  margin: 0 12px 0 0;
}

@media (min-width:0px) and (max-width:768px) {
  .c0 {
    height: auto;
  }
}

<div
  className="c0"
>
  <span
    className="c1"
    color="gray"
    size={13}
  >
    Sorting
  </span>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="c4"
      />
      <span
        className="c5"
        color="black600"
        size={14}
      >
        Price low to high
      </span>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="c4"
      />
      <span
        className="c5"
        color="black600"
        size={14}
      >
        Price high to low
      </span>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="c4"
      />
      <span
        className="c5"
        color="black600"
        size={14}
      >
        New to old
      </span>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="c4"
      />
      <span
        className="c5"
        color="black600"
        size={14}
      >
        Old to new
      </span>
    </div>
  </div>
</div>
`;
